### PR TITLE
Send virtual page views when declaration or services are completed

### PR DESCRIPTION
**Pull request by carolinegreen**
Add placeholder method for DOS until validation added in, to allow user to pass through the form
and to allow work on google analytics story.

Adding method, with tests, to track Virtual Page Views and removing G-cloud specific code to make method more generic.  This should be moved to the frontend toolkit when there is more time.

Add url to flash message to send to Google Analytics for Virtual Page Views.

Add virtual page views for 'mark as complete'.

## Fake URLs that will be sent to Google Analytics are
```
/suppliers/frameworks/digital-outcomes-and-specialists/declaration_complete
/suppliers/frameworks/digital-outcomes-and-specialists/submissions/digital-outcomes/complete
/suppliers/frameworks/digital-outcomes-and-specialists/submissions/digital-specialists/complete
/suppliers/frameworks/digital-outcomes-and-specialists/submissions/user-research-studios/complete
/suppliers/frameworks/digital-outcomes-and-specialists/submissions/user-research-participants/complete
```

### CODE REVIEW

File: app/assets/javascripts/analytics/_virtualPageViews.js [hunk 1]
```diff
@@ -1,78 +1,16 @@
...
-    'init': function () {
-      if (pathName === this.declaration.pages.firstPage) {
-        $('body').on('click', 'form.supplier-declaration fieldset:eq(0) label.selection-button', this.declaration.registerFirstQuestionInteraction);
-      }
-      if (pathName === this.createANewService.pages.firstPage) {
-        $('body').on('click', 'form fieldset label.selection-button', this.createANewService.registerLotSelection);
-      }
-    }
+  GOVUK.GDM.analytics.virtualPageViews = function() {
+    $('[data-analytics=trackPageView]').each(sendVirtualPageView);
```
Comment (robyoung) [inline app/assets/javascripts/analytics/_virtualPageViews.js]: Wow! That's ace!

**Comment by robyoung** [comment]
Looks really good. Although it might be worth squashing my WIP commit out of existence. I don't think it adds anything.

**Comment by allait** [comment]
Does this mean that we can remove `service_completed` get parameters (e.g. from https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/344/files#diff-32bf67afda65ad26faef3bf3efd522b1R304)?

**Comment by quis** [comment]
@robyoung @carolinegreen I have:
- added extra page views to send (e27f7e2)
- squashed the WIP commit

**Comment by robyoung** [comment]
Somehow it now looks like I have done all the amazing work. And then commented on my commit to say how amazing it is.

**Comment by robyoung** [comment]
:+1:

**Pull request merged**